### PR TITLE
Feat/ga/enhancements 164

### PR DESCRIPTION
**Pull request by unseen1980**
# Issue

https://github.com/SAP/guided-answers-extension/issues/164

## Description

Filter NodeEnhancements based on IDE

## Checklist for Pull Requests

- [ ] Supplied as many details as possible on this change
- [ ] Included the link to the associated issue in the Issue section above
- [ ] The code conforms to the general [development guidelines](https://github.com/SAP/guided-answers-extension/blob/main/docs/developer-guide.md).
- [ ] The code has **unit tests** where applicable and is easily unit-testable
- [ ] This branch is appropriately named for the associated issue

### CODE REVIEW

File: packages/ide-extension/src/enhancement/enhancements.ts [hunk 1]
```diff
@@ -43,17 +43,18 @@ function classifyEnhancements<T extends HTMLEnhancement | NodeEnhancement>(
     const inapplicable: T[] = [];
 
     for (const enhancement of enhancements) {
-        if (isVSCodeCommand(enhancement.command.exec)) {
-            if (
-                extensions.getExtension(enhancement.command.exec.extensionId) &&
-                enhancement.command.environment?.includes(getIde())
-            ) {
-                applicable.push(enhancement);
+        if (enhancement.command.environment.includes(getIde())) {
```
Comment (unseen1980) [inline packages/ide-extension/src/enhancement/enhancements.ts:46]: @Klaus-Keller is this what you were suggesting?

File: packages/core/test/guided-answers-api.test.ts [hunk 2]
```diff
@@ -335,7 +340,7 @@ describe('Guided Answers Api: getNodePath()', () => {
         });
 
         // Test execution
-        const result = await getGuidedAnswerApi(options).getNodePath([111, 112]);
+        const result = await getGuidedAnswerApi(options as any).getNodePath([111, 112]);
```
Comment (unseen1980) [inline packages/core/test/guided-answers-api.test.ts:343]: @Klaus-Keller I would need your help here. If I don't add the as any, then the tests fail with this. 

Screenshot 2022-08-03 at 12 23 31


Do you understand why?

Comment (Klaus-Keller) [reply packages/core/test/guided-answers-api.test.ts]: I typed the options and saw that there is a leftover in the test data: `icon`. Fixed it with commit https://github.com/SAP/guided-answers-extension/pull/198/commits/ed227e0025e829bf43632a772876a64900a1101c

**Review by Klaus-Keller — CHANGES_REQUESTED** [review]

**Review by Klaus-Keller — APPROVED** [review]

**Comment by Klaus-Keller** [comment]
@unseen1980, I've written rather lengthy review notes but somehow I managed to lose them, which rarely happens on GitHub. Anyway, in short:


https://github.com/SAP/guided-answers-extension/pull/198/commits/3466111821630803a465b27a854dc94b8c55bf46: I updated the enhancement docu to include `environment`
https://github.com/SAP/guided-answers-extension/pull/198/commits/0bd0599f76813d151f34149ce1faba11eacbbe1c: Remove link from test case
https://github.com/SAP/guided-answers-extension/pull/198/commits/f943c93c15ac6793d4f8afe4e97440649c6fea9e: Added tests for `environment.ts`, which was on my list to do and now that we also consider the environment it is good to  have them


Code looks good, thanks!

**Comment by unseen1980** [comment]
Thanks @Klaus-Keller, you rock 🤘

**Pull request merged**